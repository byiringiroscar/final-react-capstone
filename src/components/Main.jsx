import React from 'react';
import { useSelector } from 'react-redux';
import './main.css';
import Card from './Card';
import loading from '../images/loading.gif';

const Main = () => {
  const { games, isLoading, searched } = useSelector((state) => state.games);
  if (isLoading) {
    return (
      <main>
        <img className="loading-screen" src={loading} alt="loading" />
      </main>
    );
  }
  const searchedGames = games.filter((game) => game.title.toLowerCase().includes(searched.toLowerCase()) // eslint-disable-line max-len
  || game.genre.toLowerCase().includes(searched.toLowerCase())); // eslint-disable-line max-len
  return (
    <main>
      <div className="card-container">
        {searchedGames.length === 0 ? (
          <h1 className="no-result">No Results Found!</h1>
        ) : (
          searchedGames.map((item) => (
            <Card key={item.id} gameId={item.id} title={item.title} image={item.thumbnail} />
          ))
        )}
      </div>
    </main>
  );
};

export default Main;
