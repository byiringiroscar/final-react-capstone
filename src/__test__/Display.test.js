import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../redux/store';
import Main from '../components/Main';

describe('Home', () => {
  it('Render Home correctly', () => {
    const displayComponent = renderer
      .create(
        <BrowserRouter>
          <Provider store={store}>
            <Main />
          </Provider>
        </BrowserRouter>,
      )
      .toJSON();
    expect(displayComponent).toMatchSnapshot();
  });
});
