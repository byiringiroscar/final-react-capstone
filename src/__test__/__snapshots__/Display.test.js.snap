// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Render Home correctly 1`] = `
<main>
  <div
    className="card-container"
  >
    <h1
      className="no-result"
    >
      No Results Found!
    </h1>
  </div>
</main>
`;
